soft4pes/control/lin/state_space_curr_ctr.py

```diff
@@ -0,0 +1,244 @@
+
+        # Calculate the transformation angle
+        theta = np.arctan2(vg[1], vg[0])
+
+        # Get the current in dq frame
+        ic_dq = alpha_beta_2_dq(sys.x, theta)
+
+        # Maximum converter output voltage
+        u_max = conv.v_dc / 2
+
+        #Consider the filter capacitor voltage equals to the grid voltage (In case: Without considering the filter)
+        uf_dq = vg
+
+        # Compute the converter voltage reference in dq frame using the state space controller with anti-windup
+        uc_dq = self.state_space_controller(ic_dq, ic_ref_dq, uf_dq, u_max)
+
+        # Transform the converter voltage reference back to abc frame
+        uc_abc = dq_2_abc(uc_dq, theta)
+
+        u_k = uc_abc / (conv.v_dc / 2)
+
+        # Save controller data
+        ig_ref = dq_2_alpha_beta(ic_ref_dq, theta)
+        self.save_data(ig_ref, u_k, t)
+
+        # Ensure modulating signal within -1 and 1
+        return u_k
+
+    def get_state_space_ctr_pars(self):
+        """
+        Calculate state-space controller parameters.
+        
+        Returns
+        -------
+        SimpleNamespace
+            Controller parameters.
+        """
+        Ts_pu = self.Ts_pu
+        Rf = self.Rf
+        Xf = self.Xf
+
+        # Closed-loop controller bandwidth (10x crossover frequency)
+        alpha_c = 2 * np.pi / 10 / Ts_pu
+
+        # Consider delta equals to one due to not considering delay
+        delta = 1
+
+        # Coefficients
+        phi = np.exp((-Rf / Xf) * Ts_pu) * delta
+        landa = (delta - phi) / Rf
+
+        # The closed-loop pole locations
+        p_1 = 0
+        p_2 = np.exp(-alpha_c * Ts_pu)
+        p_3 = p_2
+
+        # Coefficients
+        k_2 = -p_1 - p_2 - p_3 + phi + 1
+        k_1 = (p_1 * p_2 + p_1 * p_3 + p_2 * p_3 + k_2 * phi + k_2 -
+               phi) / landa
+
+        # State feedback gain
+        K_i = np.array([k_1, 0, k_2 * 0])
+
+        # Integral gain
+        k_ii = (-p_1 * p_2 * p_3 + k_1 * landa - k_2 * phi) / landa
+
+        # Feedforward gain
+        k_ti = k_ii / (1 - p_3)
+
+        return SimpleNamespace(delta=delta, K_i=K_i, k_ii=k_ii, k_ti=k_ti)
+
+    def state_space_controller(self, ic_dq, ic_ref_dq, uf_dq, u_max):
+        """
+        State-space controller in dq frame.
+        
+        Parameters
+        ----------
+        ic_dq : 1 x 2 ndarray of floats
+            Grid Current in dq frame [p.u.].
+        ic_ref_dq : 1 x 2 ndarray of floats
+            Reference current in dq frame [p.u.].
+        uf_dq : 1 x 2 ndarray of floats
+            Grid voltage in dq frame [p.u.] (In case: Without considering the filter). 
+        u_max : float
+            Maximum converter output voltage [p.u.].        
+
+        Returns
+        -------
+        1 x 2 ndarray of floats
+            Converter voltage reference in dq frame [p.u.].
+        """
+        #In this controller uc_km1 is not considered due to not applying delay of PWM in the state-space model
```

Comment:
This comment is slightly misleading, as you assign the old value to X_LC. This comment should be where the corresponding entry of the gain matrix is set to 0.